server/src/main/java/org/opensearch/search/profile/query/ConcurrentQueryProfileBreakdown.java

```diff
@@ -47,11 +51,51 @@ public Map<String, Long> toBreakdownMap() {
         final Map<String, Long> map = new HashMap<>(super.toBreakdownMap());
 
         for (final AbstractProfileBreakdown<QueryTimingType> context : contexts.values()) {
-            for (final Map.Entry<String, Long> entry : context.toBreakdownMap().entrySet()) {
-                map.merge(entry.getKey(), entry.getValue(), Long::sum);
+            Map<String, Long> breakdown = context.toBreakdownMap();
+            for (final Map.Entry<String, Long> entry : breakdown.entrySet()) {
+                if (entry.getKey().contains(TIMING_TYPE_COUNT_SUFFIX)) {
+                    map.merge(entry.getKey(), entry.getValue(), Long::sum);
+                }
             }
+            addMaxEndTimeAndMinStartTime(map, breakdown);
         }
+        return buildQueryProfileBreakdownMap(map);
+    }
 
-        return map;
+    static void addMaxEndTimeAndMinStartTime(Map<String, Long> map, Map<String, Long> breakdown) {
+        for (QueryTimingType queryTimingType : QueryTimingType.values()) {
+            String timingType = queryTimingType.toString();
+            String timingTypeStartTime = timingType + AbstractProfileBreakdown.TIMING_TYPE_START_TIME_SUFFIX;
+            long currentMaxEndTime = map.getOrDefault(timingType + MAX_END_TIME_SUFFIX, 0L);
+            long currentMinStartTime = map.getOrDefault(timingType + MIN_START_TIME_SUFFIX, Long.MAX_VALUE);
+            // Only "create_weight" is performed at the query level
+            Map<String, Long> source = timingType.equals(QueryTimingType.CREATE_WEIGHT.toString()) ? map : breakdown;
+            long maxEndTime = Math.max(currentMaxEndTime, source.get(timingTypeStartTime) + source.get(timingType));
+            long minStartTime = Math.min(currentMinStartTime, source.get(timingTypeStartTime));
+            map.put(timingType + MAX_END_TIME_SUFFIX, maxEndTime);
+            map.put(timingType + MIN_START_TIME_SUFFIX, minStartTime);
+        }
+    }
+
+    static Map<String, Long> buildQueryProfileBreakdownMap(Map<String, Long> map) {
```

Comment:
Add some comments here too

Reply:
A javadoc has been added to this method. Thanks.